api/v1/checks.go

```diff
@@ -689,6 +689,32 @@ func (c DNSCheck) GetType() string {
 	return "dns"
 }
 
+type GCPIncident struct {
+	connection.GCPConnection `yaml:",inline" json:",inline"`
+	Subscription             string `yaml:"subscription" json:"subscription"`
+	Project                  string `yaml:"project" json:"project"`
+}
+
+type PubSubCheck struct {
+	Description  `yaml:",inline" json:",inline"`
+	Templatable  `yaml:",inline" json:",inline"`
+	GCPIncidents GCPIncident `yaml:"gcp_incidents,omitempty" json:"gcp_incidents,omitempty"`
```

Comment:
I am expecting  just connection objects to the queues here, the pubsub check itself is not aware that its being used for gcp incidents